### PR TITLE
Repair Gun Fix

### PR DESCRIPTION
**Pull request by LazDisco**
Allowed repair guns to work on the server again. For convenience, this plugin has been merged into balancemagic. Also commented out/removed a lot of unused functions.

### CODE REVIEW

File: Plugins/Public/balancemagic/Main.cpp [hunk 1]
```diff
@@ -336,141 +370,191 @@ bool ExecuteCommandString_Callback(CCmds* cmds, const wstring &wscCmd)
...
 
+/* Uneeded currently
 void __stdcall SPScanCargo(unsigned int const &iInstigatorShip, unsigned int const &iTargetShip, unsigned int iClientID)
 {
 	returncode = DEFAULT_RETURNCODE;
-}
+}*/
 
-void HkTimerCheckKick()
+// Uneeded currently
```
Comment (dsyalex) [inline Plugins/Public/balancemagic/Main.cpp]: rm

File: Plugins/Public/balancemagic/Main.cpp [hunk 2]
```diff
@@ -336,141 +370,191 @@ bool ExecuteCommandString_Callback(CCmds* cmds, const wstring &wscCmd)
...
+						CEquip *equip = eqmanager->FindByID(p1);
+						PrintUserCmdText(iDmgFrom, L"You hit a ship's possible subobj %u, and I think this is an 0x%08X of type %u.", p1, equip->archetype->iArchID, equip->GetType());
+					}
+				}
+			}
+		}*/
 	}
 }
 
+/* Uneeded currently
```
Comment (dsyalex) [inline Plugins/Public/balancemagic/Main.cpp]: rm

File: Plugins/Public/balancemagic/Main.cpp [hunk 3]
```diff
@@ -499,16 +583,16 @@ EXPORT PLUGIN_INFO* Get_PluginInfo()
...
-	p_PI->lstHooks.push_back(PLUGIN_HOOKINFO((FARPROC*)&ExecuteCommandString_Callback, PLUGIN_ExecuteCommandString_Callback, 0));
-	p_PI->lstHooks.push_back(PLUGIN_HOOKINFO((FARPROC*)&ReqModifyItem, PLUGIN_HkIServerImpl_ReqModifyItem, 0));
-	p_PI->lstHooks.push_back(PLUGIN_HOOKINFO((FARPROC*)&SPScanCargo, PLUGIN_HkIServerImpl_SPScanCargo, 0));
-	p_PI->lstHooks.push_back(PLUGIN_HOOKINFO((FARPROC*)&HkTimerCheckKick, PLUGIN_HkTimerCheckKick, 0));
-	p_PI->lstHooks.push_back(PLUGIN_HOOKINFO((FARPROC*)&Plugin_Communication_Callback, PLUGIN_Plugin_Communication, 10));
+	p_PI->lstHooks.emplace_back(reinterpret_cast<FARPROC*>(&LoadSettings), PLUGIN_LoadSettings, 0);
+	p_PI->lstHooks.emplace_back(reinterpret_cast<FARPROC*>(&HkCb_AddDmgEntry), PLUGIN_HkCb_AddDmgEntry, 0);
+	p_PI->lstHooks.emplace_back(reinterpret_cast<FARPROC*>(&UserCmd_Process), PLUGIN_UserCmd_Process, 0);
+	p_PI->lstHooks.emplace_back(reinterpret_cast<FARPROC*>(&ExecuteCommandString_Callback), PLUGIN_ExecuteCommandString_Callback, 0);
+	//p_PI->lstHooks.emplace_back(reinterpret_cast<FARPROC*>(&ShipDestroyed), PLUGIN_ShipDestroyed, 0);
```
Comment (dsyalex) [inline Plugins/Public/balancemagic/Main.cpp]: rm these

**Comment by dsyalex** [comment]
There are still some reasons to do this commit, though I did re-make the config file that went missing for a while based on the logic used in the code before it got moved to config:
```
[HealingRate]
target_shipclass = 1
target_shipclass = 3
addition = 600

[HealingRate]
target_shipclass = 2
target_shipclass = 4
target_shipclass = 5
addition = 1200

[HealingRate]
target_shipclass = 6
target_shipclass = 7
target_shipclass = 8
target_shipclass = 9
target_shipclass = 10
target_shipclass = 11
target_shipclass = 12
multiplier = 0.75
addition = 2400

[HealingRate]
target_shipclass = 13
target_shipclass = 14
target_shipclass = 15
multiplier = 0.5
addition = 3600

[HealingRate]
target_shipclass = 16
target_shipclass = 17
target_shipclass = 18
multiplifer = 0.2
addition = 12000
```

**Comment by LazDisco** [comment]
I have deleted my FLHook fork and am now unable to make changes. Closing request.

**Pull request closed without merging**